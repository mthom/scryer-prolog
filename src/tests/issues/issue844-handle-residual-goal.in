use_module(library(dif)).
use_module(library(atts)).
-X\=X.
-X=X.
dif(-X,X).
dif(-X,X), -X=X.
call_residue_vars(dif(-X,X), Vars).
set_prolog_flag(occurs_check, true).
-X\=X.
dif(-X,X).
halt.
