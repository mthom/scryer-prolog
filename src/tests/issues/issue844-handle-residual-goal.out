   true.
   true.
   false.
   X = -X.
   dif:dif(-X,X).
   false.
   Vars = [X], dif:dif(-X,X).
   true.
   true.
   true.
